scripts/parallel-test.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e # exit on failure of any "simple" command (excludes &&, ||, or | chains)
+# prepare environment
+PATH=$PATH:~/opt/mongodb/bin
+echo "Extracting build directory..."
+tar -zxf build.tar.gz
+echo "Starting MongoDB..."
+~/bin/mongod --fork --dbpath ~/data/mongodb -f ~/etc/mongod.conf --logpath "$(pwd)"/mongod.log
+cd /data/job/build
+# run tests
+echo "Running tests..."
+CPU_CORES=$(getconf _NPROCESSORS_ONLN)
+echo "$CPU_CORES cpu cores detected."
+TEST_COUNT=$(ctest -N -LE _tests | grep -i 'Total Tests: ' | cut -d ':' -f 2 | awk '{print $1}')
+[[ $TEST_COUNT > 0 ]] && echo "$TEST_COUNT tests found." || (echo "ERROR: No tests registered with ctest! Exiting..." && exit 1)
+echo "$ ctest -j $CPU_CORES -LE _tests --output-on-failure -T Test"
```

Comment:
If you want to print the command you're running and avoid duplication (which is at risk of diverging over time), one approach is

`(set -x; echo hello)`

Reply:
Unfortunately I did not see this comment before I merged, I will be more cautious in the future!

I agree with you that this would be a better approach in these test scripts. I ran this on my local machine in both iterm2 and terminal, and it didn't have the expected behavior that you and others online described. Both "set -x" and "set -v" print a bunch of junk, looks like a stack trace, on my local machine. This undesired behavior must be a mac thing because both worked as expected in a linux docker image.

Either way, we will be doing more work on this pipeline soon and I have added this to my notes. I will implement a platform-independent method of echoing desired commands without duplication next time I work on this pipeline.